### PR TITLE
65012 custom date advanced search

### PR DESCRIPTION
**Pull request by fazilqa**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

## Summary

- This PR adds an E2E scenarios to verify advanced filter custom date range feature behavior for all folders. Updates constants, parametrizes methods and renames the existing tests accordingly to general naming pattern.

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-65012

## Testing done

- Relevant tests have been executed in headed mode with a 100x pattern.
- All tests have been successfully conducted in headless mode without any failures.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/tests/e2e/pages/PatientSearchPage.js [hunk 1]
```diff
@@ -137,15 +137,113 @@ class PatientSearchPage {
           const extractedDate = dateString.split(' at ')[0]; // "November 29, 2024"
           const parsedDate = new Date(extractedDate);
 
-          // calculate three months back from the current date
-          const threeMonthsBack = new Date();
-          threeMonthsBack.setMonth(threeMonthsBack.getMonth() - numberOfMonth);
+          // calculate a few months back from the current date
+          const currentDate = new Date();
+          currentDate.setMonth(currentDate.getMonth() - numberOfMonth);
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/pages/PatientSearchPage.js:142]: aren't you already using a function `getRandomDateWithinLastNumberOfMonths` for this?

File: src/applications/mhv-secure-messaging/tests/e2e/sort-filter-test/secure-messaging-advanced-filter-custom-date-custom-folder.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      `any`,
+    );
+    cy.get(Locators.BLOCKS.FILTER_START_DATE).should(`not.exist`);
+    cy.get(Locators.BLOCKS.FILTER_END_DATE).should(`not.exist`);
+
+    cy.injectAxe();
+    cy.axeCheck(AXE_CONTEXT);
+  });
+
+  it('verify search results', () => {
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/sort-filter-test/secure-messaging-advanced-filter-custom-date-custom-folder.cypress.spec.js:89]: linter is warning about axe check in this test

File: src/applications/mhv-secure-messaging/tests/e2e/sort-filter-test/secure-messaging-advanced-filter-inbox.cypress.spec.js [hunk 3]
```diff
@@ -48,6 +48,10 @@ describe('SM INBOX ADVANCED FIXED DATE RANGE SEARCH', () => {
 
     PatientSearchPage.verifySearchResponseLength(searchResultResponse);
     PatientSearchPage.verifyMessageDate(3);
+    PatientSearchPage.verifySearchMessageLabel(
+      searchResultResponse,
+      'Last 3 months',
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/sort-filter-test/secure-messaging-advanced-filter-inbox.cypress.spec.js:53]: Looks like you repeatedly use the same string values for validations. Would it be more convenient to turn int to a constant and use a variable across?

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
@fazilqa I left couple comments below. other than that - looks great

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**